### PR TITLE
Model replacement

### PR DESCRIPTION
**Pull request by KirillDogadin-std**

### CODE REVIEW

File: code/MultiKE_model_torch.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+from utils import *
+
+
+def xavier_init(dim1, dim2, is_l2_norm):
+    ret = torch.empty(dim1, dim2)
+    ret = nn.Parameter(
+        nn.init.xavier_normal_(ret)
+    )
+    if is_l2_norm:
+        ret = nn.functional.normalize(ret, dim=1, p=2)
```
Comment (KirillDogadin-std) [inline code/MultiKE_model_torch.py:10]: L2 normalize should be done in forward-pass (forward method of class nn.Module) not during init because of the dynamic nature of graph in Pytorch.

Comment (KirillDogadin-std) [reply code/MultiKE_model_torch.py:10]: @arashamini on the second thought, having that in the init function of tensorflow 
```python
def xavier_init(shape, name, is_l2_norm, dtype=None):
    with tf.name_scope('xavier_init'):
        embeddings = tf.get_variable(name, shape=shape, dtype=dtype,
                                     initializer=tf.contrib.layers.xavier_initializer(uniform=False))
    return tf.nn.l2_normalize(embeddings, 1) if is_l2_norm else embeddings
```

still makes me wonder why one needs to replace the l2 so that the corresponding module ends up in `forward`. Since from `tf` snippet it appears to me that the `l2` is applied once and that's it. If I understand correctly `forward` in `torch` will apply `l2` on each input of the info.

**Comment by KirillDogadin-std** [comment]
@4rash-uni 
I assume you refer to that line 33. Is that correct?

In github you can add comments to specific lines via `files changed`  tab by clicking on a plus to the left of the line or by click&drag to select multiple.

image

**Comment by KirillDogadin-std** [comment]
So currently seems that the whole file was mapped.

- flaws expected
- several parts were not tested like `train` funcions because I didnt manage to set up such a dataset which will make code run that far. Dataset I tried to use can be found in https://github.com/KirillDogadin-std/Lab-Semantic/tree/smaller-data.

Small info on how it was composed:
```
    - grab first lines of kg1 (entities),
    - based on these fetch the rest of the info from the other files
    - via file that bonds these datasets fetch the entities that are related in the other graph.
    - having infos about the entities in the other graph fetch the info from the other kg2 desctiptor-files.
```

- a lot of templates, while reviewing, one can pay attention to:
   - single `define....graph`, call it `a` for example
   - single `train` that calls `a` inside.
   - some loss function
   - `get_optimizer()`
   - `_define_variables`
   - _optionally_: `def conv(..):`
